pkg/webhook/webhook.go

```diff
@@ -42,49 +43,33 @@ func (p *Webhook) acceptHeaderCheck(w http.ResponseWriter, r *http.Request) erro
 }
 
 func (p *Webhook) headerCheck(isContentType bool, w http.ResponseWriter, r *http.Request) error {
-	var header string
+	header := acceptHeader
 	if isContentType {
-		header = r.Header.Get(contentTypeHeader)
-	} else {
-		header = r.Header.Get(acceptHeader)
+		header = contentTypeHeader
 	}
 
-	if len(header) == 0 {
+	v := r.Header.Get(header)
+
+	if len(v) == 0 {
 		w.Header().Set(contentTypeHeader, contentTypePlaintext)
 		w.WriteHeader(http.StatusNotAcceptable)
 
-		msg := "client must provide "
-		if isContentType {
-			msg += "a content type"
-		} else {
-			msg += "an accept header"
-		}
-		err := fmt.Errorf(msg)
+		err := fmt.Errorf("client must provide %s header", header)
+
+		io.WriteString(w, err.Error())
 
-		_, writeErr := fmt.Fprint(w, err.Error())
-		if writeErr != nil {
-			requestLog(r).WithField(logFieldError, writeErr).Fatalf("error writing error message to response writer")
-		}
 		return err
 	}
 
 	// as we support only one media type version, we can ignore the returned value
-	if _, err := checkAndGetMediaTypeHeaderValue(header); err != nil {
+	if _, err := checkAndGetMediaTypeHeaderValue(v); err != nil {
 		w.Header().Set(contentTypeHeader, contentTypePlaintext)
 		w.WriteHeader(http.StatusUnsupportedMediaType)
 
-		msg := "client must provide a valid versioned media type in the "
-		if isContentType {
-			msg += "content type"
-		} else {
-			msg += "accept header"
-		}
+		err := fmt.Errorf("client must provide a valid versioned media type in the %s header: %w", header, err)
+
+		io.WriteString(w, err.Error())
```

Comment:
for reviewers, it's convention to ignore errors like this as they don't matter and there's nothing which can be done about them.